core/src/test/java/org/elasticsearch/search/profile/TimerTests.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.search.profile;
+
+import org.elasticsearch.test.ESTestCase;
+
+import java.util.concurrent.atomic.AtomicLong;
+
+public class TimerTests extends ESTestCase {
+
+    public void testTimingInterval() {
+        final AtomicLong counter = new AtomicLong();
+        Timer t = new Timer() {
+            long time = 50;
+            @Override
+            long nanoTime() {
+                counter.incrementAndGet();
+                return time += 1;
+            }
+        };
+        for (int i = 0; i < 100000; ++i) {
```

Comment:
nit: unrelated but maybe we should use a separator here and write it as `100_000`?